components/post_view/post_info/post_info.jsx

```diff
@@ -292,14 +292,15 @@ export default class PostInfo extends React.PureComponent {
         const fromAutoResponder = PostUtils.fromAutoResponder(post);
 
         let flagIcon;
-        if (!isEphemeral && !post.failed && !isSystemMessage && (this.props.hover || this.props.isFlagged)) {
+        if (!isEphemeral && !post.failed && !isSystemMessage) {
             flagIcon = (
                 <PostFlagIcon
                     idPrefix='centerPostFlag'
                     idCount={idCount}
                     postId={post.id}
                     isFlagged={this.props.isFlagged}
                     isEphemeral={isEphemeral}
+                    style={{visibility: this.props.hover || this.props.isFlagged ? 'hidden' : 'visible'}}
```

Comment:
@saturninoabril Can we do this with `display: none` for toggling?.

We are not rendering the component if it not hovered so we skip mounting components. I think this was part of previous optimisations for postList to not render components for postList mount. 

With `visibility` we will be mounting and are hidden just by css so, it can cost us on mount.

Reply:
It's not working with `display: none`.
On current implementation, I agree on performance optimisation on channel switching but we have wasteful rendering on hover or mouseover during scrolling, which I hope the CSS should do the trick instead of javascript logic.

Anyway, let me take a look again at this after completing MM-13434 which aims to improve performance of PostHeader.  I'll gather performance data first to back up this change.

Reply:
We should be more concerned about the optimisation for mount of 30 posts i.e a channel switch or load more messages because that is where we see a bigger impact than on hover of scroll which is re-rendering one component.

I don't know the impact of this but while you are at the other ticket it would be great if you could cherry-pick this and test it with that. Let me know if it does not make much of difference

Reply:
@sudheerDev I don't see much of a difference on channel switch.  Please see below comparison.  Also note that 2 of our contributors are also experiencing the same issue of this kind on hover (as discussed in community server, [1](https://community.mattermost.com/core/pl/kzjfth9e7pgwz8jsne94up34pc), [2](https://community.mattermost.com/core/pl/rhes6ysopfnejxerz4xpx7z7ga)).  I suggested this approach as it's the the only way I know at the moment on how we can test post header (dot menu, timestamp, flag, reaction).  I'd appreciate to learn other solution though.

Channel switching on this PR:
<img width="1385" alt="mm-13222_channel_switch" src="https://user-images.githubusercontent.com/5334504/52197227-b6977f80-2899-11e9-96ca-5049fb174372.png">

Channel switching on master:
<img width="1589" alt="master_channel_switch" src="https://user-images.githubusercontent.com/5334504/52197207-a67fa000-2899-11e9-9b82-e4660a946f8d.png">

Reply:
Probably the difference with the flag icon and the date is not big, and I'm ok with the css approach, but for the dots menu, renderering it only on hover have a huge performance improvement. Actually we didn't doing that for the search results and was super slow.

Probably we have to think more about this.

Reply:
@saturninoabril @jespino Ya dot menu is an expensive component. Looks like the components here are not that expensive.

Reply:
Thanks @saturninoabril